samples/WinForms/MainFormWinFormsDx12.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright © Tanner Gooding and Contributors. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+
+using System;
+using System.Windows.Forms;
+
+namespace TerraFX.Samples.WinForms
+{
+    /// <summary>
+    /// Main form of the project TerraFX.Samples.WinForms.
+    /// It is a WinForms GUI with a Panel that contains DirectX content.
+    /// The DirectX content is taken from the Hello Triangle example.
+    /// The main point is to demonstrate how to connect a WinForms GUI to DirectX rendering.
+    /// This connection is encapsulated in the DX12Panel class.
+    /// </summary>
+    public partial class MainFormWinFormsDX12 : Form
+    {
+        private readonly DX12Panel _panel;
+
+        public MainFormWinFormsDX12()
+        {
+            InitializeComponent();
+            _panel = InitDX12Panel();
+        }
+
+        private DX12Panel InitDX12Panel()
+        {
+            var dxPanel = new DX12Panel(MainSplitter.Panel2.Width, MainSplitter.Panel2.Height);
+            MainSplitter.Panel2.Controls.Add(dxPanel);
+            dxPanel.Dock = DockStyle.Fill;
+            return dxPanel;
+        }
+
+        private void ColorBtn_Click(object sender, EventArgs e) => _panel.OnRender();
```

Comment:
I don't think this is layered correctly. The panel should self render at the appropriate frame rate, not on demand.

Reply:
No its correct. While there is no change to the graphics scene there should be no re-render. Compute resources should be spent only when needed.

Reply:
That's not how multimedia panels typically work. By default, you render at a given FPS and you only fallback to on demand if requested. Otherwise animations and the like don't work.

Reply:
This is especially true for the application I have in mind at the end. There the rendering will be so expensive that it can not be done at an appropriate frame rate. Hence I'll have to simplify the scene while the user interacts with it and render the full quality when interaction is over (upon mouse up event).
This has no equivalent in a game. There everything must be same quality all the time.

Reply:
hmm, I can add a refresh timer to render an a particular frame rate. But then something has to change for it to be visual.   Let's keep this PR as is. Then I'll make another to port the HelloConstantBuffer to TerraFX.Interop.Windows.sln. When that works I'll reapply it to the WinForms project with a timer that calls OnRender at 60Hz. But I'd really like that to come from a new branch PR, not this one that is on the master.

Reply:
Sure, but this sample isn't for your game, its for the general audience coming to TerraFX.Interop.Windows.

I'm fine with the sample panel having a built in mechanism to specify the target framerate and even for that to be `0` so that you can make it explicit. But the default should likely just be the same as the rest of the samples.

Reply:
ok

Reply:
Logged issue https://github.com/terrafx/terrafx.interop.windows/issues/62 so we do not forget about this.  I'll implement after HelloConstantBuffer  is working. So no further change needed under this PR.